### ISSUE TITLE
sparklyr with h2o - context creation issue 

### ISSUE THREAD

**Issue by ganeshmailbox** [rstudio/sparklyr#237]
I have a problem creating a spark context with the sparkling water, steps as given in the sparklyr documentation. This is what i tried: 

```
library(sparklyr)
library(rsparkling)
library(dplyr)

sc <- spark_connect("local", version = "1.6.2")
```

moment i add the library `rsparkling` and try to create spark connect, i am getting an error as below. No issues if i tried to connect without loading rsparkling. 

```
Error in start_shell(master = master, spark_home = spark_home, spark_version = version,  : 
  Failed to launch Spark shell. Ports file does not exist.
    Path: C:\Users\<user>\AppData\Local\rstudio\spark\Cache\spark-1.6.2-bin-hadoop2.6\bin\spark-submit.cmd
    Parameters: --class, sparklyr.Backend, --packages, "com.databricks:spark-csv_2.11:1.3.0","com.amazonaws:aws-java-sdk-pom:1.10.34","ai.h2o:sparkling-water-core_2.10:1.6.7","ai.h2o:sparkling-water-ml_2.10:1.6.7","ai.h2o:sparkling-water-repl_2.10:1.6.7", "C:\Programs\R\R-3.3.1\library\sparklyr\java\sparklyr-1.6-2.10.jar", C:\Users\<user>\AppData\Local\Temp\RtmpEds4nB\file2825066c92978.out

Ivy Default Cache set to: C:\Users\<user>\.ivy2\cache
The jars for the packages stored in: C:\Users\<user>\.ivy2\jars
:: loading settings :: url = jar:file:/C:/Users/<user>/AppData/Local/rstudio/spark/Cache/spark-1.6.2-bin-hadoop2.6/lib/spark-assembly-1.6.2-hadoop2.6.0.jar!/org/apache/ivy/core/settings/ivysettings.xml
com.databricks#spark-csv_2.11 added as a dependency
com.amazonaws#aws-java-sdk-pom add
```

I tried with and without http_proxy in the environment, same result. Also tried to connect with one config entry NULL, but got the same error. **Please let know if you have any solution to this problem.**

```
config <- spark_config()
config[["sparklyr.defaultPackages"]] <- NULL
sc <- spark_connect(master = "local", config = config)

```

The session info is attached below

```
> sessionInfo()
R version 3.3.1 (2016-06-21)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1
locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    
attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     
other attached packages:
[1] dplyr_0.5.0      rsparkling_0.1.0 h2o_3.10.0.6     statmod_1.4.26   sparklyr_0.4    
loaded via a namespace (and not attached):
 [1] Rcpp_0.12.6        knitr_1.14         magrittr_1.5       rappdirs_0.3.1     R6_2.1.3           stringr_1.1.0      tools_3.3.1       
 [8] parallel_3.3.1     config_0.2         DBI_0.5-1          withr_1.0.2        htmltools_0.3.5    yaml_2.1.13        assertthat_0.1    
[15] rprojroot_1.0-2    digest_0.6.10      tibble_1.2         formatR_1.4        bitops_1.0-6       RCurl_1.95-4.8     evaluate_0.9 
```

**Comment by javierluraschi** [comment]
@ganeshmailbox which version of sparklyr? Could you upgrade to the latest version using: `devtools::install_github("rstudio/sparklyr")` and share again the log? There are a couple fixes in `0.4.1` that improve logging while connecting. Thanks!

**Comment by javierluraschi** [comment]
`Ports file does not exist.` error has been solved for many cases, it is likely that this won't reproduce in latest version of `sparklyr`.
